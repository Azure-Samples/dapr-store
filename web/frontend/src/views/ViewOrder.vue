<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Show a single order
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <h1>
      View Order Status
      <b-button size="lg" variant="success" class="float-right" @click="loadOrder($route.params.id)">
        <fa icon="redo-alt" /> &nbsp; Refresh
      </b-button>
    </h1>

    <error-box :error="error" />

    <div v-if="!order && !error" class="text-center">
      <b-spinner variant="success" style="width: 5rem; height: 5rem;" />
    </div>

    <order v-if="order" :order="order" />
  </div>
</template>

<script>
import ErrorBox from '../components/ErrorBox'
import api from '../services/api'
import Order from '../components/Order'

export default {
  name: 'ViewOrder',

  components: {
    'error-box': ErrorBox,
    'order': Order
  },

  data() {
    return {
      order: null,
      error: null
    }
  },

  mounted() {
    this.loadOrder(this.$route.params.id)
  },

  methods: {
    async loadOrder(id) {
      this.order = null
      try {
        this.order = await api.orderGet(id)
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>
