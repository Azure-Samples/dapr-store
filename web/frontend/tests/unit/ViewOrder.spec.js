import { createLocalVue, mount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import BootstrapVue from 'bootstrap-vue'
const localVue = createLocalVue()
localVue.use(BootstrapVue)

import VueRouter from 'vue-router'
localVue.use(VueRouter)
const router = new VueRouter()

import ViewOrder from '@/views/ViewOrder.vue'

jest.mock('@/services/api')

const orderId = 'order123'

describe('ProductSingle.vue', () => {
  it('renders product details', async () => {
    router.push({ name: 'view-order', params: { id: orderId } })

    const wrapper = mount(ViewOrder, {
      localVue,
      router
    })

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})