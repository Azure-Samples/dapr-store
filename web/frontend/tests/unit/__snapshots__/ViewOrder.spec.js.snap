// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSingle.vue renders product details 1`] = `
<div>
  <h1>
    View Order Status
    <button type="button" class="btn float-right btn-success btn-lg">
      <fa icon="redo-alt"></fa> &nbsp; Refresh
    </button></h1>
  <!---->
  <div class="text-center"><span aria-hidden="true" class="spinner-border text-success" style="width: 5rem; height: 5rem;"><!----></span></div>
  <!---->
</div>
`;
